bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/handler/HaywardBridgeHandler.java

```diff
@@ -0,0 +1,458 @@
+                if (handler != null) {
+                    handler.getTelemetry(xmlResponse);
+                }
+            }
+        }
+        return true;
+    }
+
+    public synchronized boolean getAlarmList() throws Exception {
+        for (Thing thing : getThing().getThings()) {
+            Map<String, String> properties = thing.getProperties();
+            if (properties.get(HaywardBindingConstants.PROPERTY_TYPE).equals("BACKYARD")) {
+                HaywardBackyardHandler handler = (HaywardBackyardHandler) thing.getHandler();
+                if (handler != null) {
+                    String systemID = properties.get(HaywardBindingConstants.PROPERTY_SYSTEM_ID);
+                    if (systemID != null) {
+                        return handler.getAlarmList(systemID);
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private synchronized void initPolling(int initalDelay) {
+        pollTelemetryFuture = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                if (commFailureCount >= 5) {
+                    commFailureCount = 0;
+                    clearPolling(pollTelemetryFuture);
+                    clearPolling(pollAlarmsFuture);
+                    initialize();
+                    return;
+                }
+                if (!(getTelemetryData())) {
+                    commFailureCount++;
+                    return;
+                }
+            } catch (Exception e) {
+                logger.debug("Hayward Connection thing: Exception during poll: {}", e.getMessage());
+            }
+        }, initalDelay, config.telemetryPollTime, TimeUnit.SECONDS);
+        return;
+    }
+
+    private synchronized void initAlarmPolling(int initalDelay) {
+        pollAlarmsFuture = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                getAlarmList();
+            } catch (Exception e) {
+                logger.debug("Hayward Connection thing: Exception during poll: {}", e.getMessage());
+            }
+        }, initalDelay, config.alarmPollTime, TimeUnit.SECONDS);
+    }
+
+    private void clearPolling(@Nullable ScheduledFuture<?> pollJob) {
+        if (pollJob != null) {
+            pollJob.cancel(false);
+        }
+    }
+
+    @Nullable
+    Thing getThingForType(HaywardTypeToRequest type, int num) {
+        for (Thing thing : getThing().getThings()) {
+            Map<String, String> properties = thing.getProperties();
+            if (properties.get(HaywardBindingConstants.PROPERTY_SYSTEM_ID).equals(Integer.toString(num))) {
+                if (properties.get(HaywardBindingConstants.PROPERTY_TYPE).equals(type.toString())) {
+                    return thing;
+                }
+            }
+        }
+        return null;
+    }
+
+    public List<String> evaluateXPath(String xpathExp, String xmlResponse) throws HaywardException {
```

Comment:
This exception is never thrown.
```suggestion
    public List<String> evaluateXPath(String xpathExp, String xmlResponse) {
```